### PR TITLE
Setup proper routing for table ajax

### PR DESCRIPTION
**Pull request by felorhik**
I noticed when I was starting out with this package that the ajax source for datatable is always the root directory and there is no method to override this without editing the default view.

Made an update here that updates the view files allowing you to use `$table->option('route', URL_HERE)` to define the ajax source.

Not sure if this is the best way to do it since from what I see of this package so far it assumes that the ajax source is from the same route as the view source.

EDIT:

Noticed that the columns were based on the value of the array rather then the index, when the value is the label. This leads to errors when overriding table headers.

### CODE REVIEW

File: src/views/datatable110.blade.php [hunk 1]
```diff
@@ -4,12 +4,12 @@
         oTable = jQuery('#{{ $id }}').DataTable({
             "processing": true,
             "serverSide": true,
-            "ajax": "/",
+            "ajax": "{{ isset($options['route']) ? $options['route'] : '/' }}",
             "columns": [
                 @foreach($columns as $name => $label)
-                { 'data': '{{ $label }}' },
+                { 'data': '{{ $name }}' },
```
Comment (Chumper) [inline src/views/datatable110.blade.php:10]: Thanks for the contribution!
Why did you change from `label` to `name`?

File: src/views/datatable110.blade.php [hunk 2]
```diff
@@ -4,12 +4,12 @@
         oTable = jQuery('#{{ $id }}').DataTable({
             "processing": true,
             "serverSide": true,
-            "ajax": "/",
+            "ajax": "{{ isset($options['route']) ? $options['route'] : '/' }}",
```
Comment (Chumper) [inline src/views/datatable110.blade.php:7]: We should add a separate option for this instead of putting this into the options.
The options are normally used for something else that is not implemented yet :/

Comment (felorhik) [inline src/views/datatable110.blade.php:7]: I agree but since option was there I thought it might be the quickest way to solve it at least in the context of the default template.

File: src/views/datatable110.blade.php (hunk 1 above)
Comment (felorhik) [inline src/views/datatable110.blade.php:10]: When its going through the data it uses this to define what the index's of the data is, so when you use label it looks for lets say `#` for id instead of `id`. Changing it fixes it so that label is only used in the header and the data properly maps to the right columns.

Comment (Chumper) [inline src/views/datatable110.blade.php:10]: yeah, correct, makes sense now :)

**Comment by Chumper** [comment]
Will merge it in but will remove the route option later when we introduce a separate option.

**Comment by timgws** [comment]
@Chumper what would you like to call the option?

I am thinking something along the lines of:

``` php
        $t = Datatable::make(new CollectionProvider(Categories::all()))
            ->column('id') // show the id column of the user model
            ->column('name', null, Searchable::NONE(), Orderable::NONE())
            ->endpoint('admin.categories.datatable')
            ->build();
```

Where `endpoint()` would call the `route()` function to find out the final URL to send AJAX requests to.

**Comment by Chumper** [comment]
IMO this should be an option on the view table object, as the backend does not care about any url, as it could be requested from any url.

So i would propose this in the view:

``` php
{{
    $datatable
        ->headers() // tell the table to render the header in the table
        ->columns('id', '#') // show # in the header instead of 'id'
        ->columns('name', 'Full name') // show 'Full name' in the header instead of 'name'
        ->endpoint('asd.asd')
        ->table(); // render just the table
}}
{{
    $datatable
        ->script() // now render the script
}
```

So it will be a member of https://github.com/OpenSkill/Datatable/blob/master/src/OpenSkill/Datatable/Views/DatatableView.php

**Comment by timgws** [comment]
OK, but there is only one small issue here. The `$datatable` object is called twice, it will mean that you might need to change calling `->endpoint()` if you move the `script()` before the `table()`.

I am thinking that calling `->endpoint()` will store the variable in a private property, so it can be accessed by whatever needs to use it, but it just means the developer will need to ensure that `->endpoint()` is called the first time that it is used when rendering the datatable for the first time.

This is why I think that it might be better to have it a little further up (however, doing that means editing OpenSkill\Datatable\Datatable & OpenSkill\Datatable\DatatableService and a whole bunch of other dependancies, which breaks the single responsibility principle).
